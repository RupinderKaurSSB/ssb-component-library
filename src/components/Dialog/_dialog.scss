.ssb-dialog {
  @include open-sans;
  display: flex;
  margin: 20px;
  max-width: 880px;
  min-height: 95px;

  .icon-panel {
    color: $ssb-white;
    display: flex;
    flex-shrink: 0;
    font-size: 36px;
    justify-content: center;
    padding-top: 20px;
    width: 90px;

    .icon {
      font-size: 40px;
    }
  }

  .dialog-content {
    padding: 16px;

    .dialog-title {
      font-weight: bold;
    }

    .content {
      margin-top: 8px;
    }
  }

  &.warning {
    border: 2px solid $ssb-red-3;

    .icon-panel {
      background: $ssb-red-3;
    }
  }

  &.info {
    border: 2px solid $ssb-blue-3;

    .icon-panel {
      background: $ssb-blue-3;
    }
  }
}
