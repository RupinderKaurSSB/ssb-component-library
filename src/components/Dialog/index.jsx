import React from 'react';
import PropTypes from 'prop-types';
import Paragraph from '../Paragraph';

const Dialog = ({ children, title, type }) => (
	<div className={`ssb-dialog ${type}`}>
		<div className="icon-panel">
			{type === 'warning' && <i className="feather-alert-triangle" />}
			{type === 'info' && <i className="feather-alert-circle" />}
		</div>
		<div className="dialog-content">
			<span className="dialog-title">{title}</span>
			<Paragraph>{children}</Paragraph>
		</div>
	</div>
);

Dialog.defaultProps = {
	type: 'info',
};

Dialog.propTypes = {
	children: PropTypes.node.isRequired,
	title: PropTypes.string.isRequired,
	type: PropTypes.oneOf([
		'info',
		'warning',
	]),
};

export default Dialog;
