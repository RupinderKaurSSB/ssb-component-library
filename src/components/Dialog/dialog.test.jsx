import React from 'react';
import {shallow} from 'enzyme';
import Dialog from './index';

describe('Dialog component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Dialog	title="Dialog Title">Dialog</Dialog>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Change dialog type Toggles classNames correctly', () => {
		const wrapper = shallow(<Dialog	title="Dialog Title" type="warning">Dialog</Dialog>);
		console.log(wrapper.debug())
		expect(wrapper.hasClass('warning')).toEqual(true);
	});
});
