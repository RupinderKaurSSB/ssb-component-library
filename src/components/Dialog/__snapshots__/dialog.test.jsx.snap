// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component Matches the snapshot 1`] = `
<div
  className="ssb-dialog info"
>
  <div
    className="icon-panel"
  >
    <Info
      className="icon"
      color="currentColor"
      size={45}
    />
  </div>
  <div
    className="dialog-content"
  >
    <span
      className="dialog-title"
    >
      Dialog Title
    </span>
    <div
      className="content"
    >
      Dialog
    </div>
  </div>
</div>
`;
